pkg/polaris/graphql/core/core.go

```diff
@@ -48,53 +50,137 @@ const (
+type PermissionGroup string
 
 const (
-	FeatureInvalid                       Feature = ""
-	FeatureAll                           Feature = "ALL"
-	FeatureAppFlows                      Feature = "APP_FLOWS"
-	FeatureArchival                      Feature = "ARCHIVAL"
-	FeatureAzureSQLDBProtection          Feature = "AZURE_SQL_DB_PROTECTION"
-	FeatureAzureSQLMIProtection          Feature = "AZURE_SQL_MI_PROTECTION"
-	FeatureCloudAccounts                 Feature = "CLOUDACCOUNTS" // Deprecated, no replacement.
-	FeatureCloudNativeArchival           Feature = "CLOUD_NATIVE_ARCHIVAL"
-	FeatureCloudNativeArchivalEncryption Feature = "CLOUD_NATIVE_ARCHIVAL_ENCRYPTION"
-	FeatureCloudNativeBLOBProtection     Feature = "CLOUD_NATIVE_BLOB_PROTECTION"
-	FeatureCloudNativeProtection         Feature = "CLOUD_NATIVE_PROTECTION"
-	FeatureCloudNativeS3Protection       Feature = "CLOUD_NATIVE_S3_PROTECTION"
-	FeatureExocompute                    Feature = "EXOCOMPUTE"
-	FeatureGCPSharedVPCHost              Feature = "GCP_SHARED_VPC_HOST"
-	FeatureServerAndApps                 Feature = "SERVERS_AND_APPS"
-	FeatureRDSProtection                 Feature = "RDS_PROTECTION"
-	FeatureKubernetesProtection          Feature = "KUBERNETES_PROTECTION"
+	PermissionGroupInvalid           PermissionGroup = "GROUP_UNSPECIFIED"
+	PermissionGroupBasic             PermissionGroup = "BASIC"
+	PermissionGroupRSCManagedCluster PermissionGroup = "RSC_MANAGED_CLUSTER"
 )
 
-var validFeatures = map[Feature]struct{}{
-	FeatureAll:                           {},
-	FeatureAppFlows:                      {},
-	FeatureArchival:                      {},
-	FeatureAzureSQLDBProtection:          {},
-	FeatureAzureSQLMIProtection:          {},
-	FeatureCloudAccounts:                 {},
-	FeatureCloudNativeArchival:           {},
-	FeatureCloudNativeArchivalEncryption: {},
-	FeatureCloudNativeBLOBProtection:     {},
-	FeatureCloudNativeProtection:         {},
-	FeatureCloudNativeS3Protection:       {},
-	FeatureExocompute:                    {},
-	FeatureGCPSharedVPCHost:              {},
-	FeatureKubernetesProtection:          {},
-	FeatureRDSProtection:                 {},
-	FeatureServerAndApps:                 {},
+// Feature represents a Polaris cloud account feature with a set of permission
+// groups. If the PermissionGroups field is nil then the full set of permissions
+// are used for the feature.
+type Feature struct {
+	Name             string            `json:"featureType"`
+	PermissionGroups []PermissionGroup `json:"permissionsGroups"`
+}
+
+// Equal returns true if the features have the same name. Note, this function
+// does not compare the permission groups.
+func (feature Feature) Equal(other Feature) bool {
+	return feature.Name == other.Name
+}
+
+// DeepEqual returns true if the features are equal. The features are equal if
+// they have the same name and the same permission groups.
+func (feature Feature) DeepEqual(other Feature) bool {
```

Comment:
nit: duplicate permission groups will cause a false positive, e.g. `equal` is `true` if:
```
    f := Feature{
        PermissionGroups: []PermissionGroup{
            PermissionGroupBasic,
            PermissionGroupBasic,
        },
    }

    equal := f.DeepEqual(Feature{
        PermissionGroups: []PermissionGroup{
            PermissionGroupBasic,
            PermissionGroupRSCManagedCluster,
        },
    })
```